### ISSUE TITLE
Expose shape function in the cupy top level namespace

### ISSUE THREAD

**Issue by beckernick** [cupy/cupy#3847]
For compatibility with NumPy, it would be valuable to expose `ndarray.shape` as a module level function.

```python
import cupy as cp
import numpy as np
​
arr = np.arange(5)
np.shape(arr) == arr.shape
True
```

The NumPy approach to this looks quite reasonable: https://github.com/numpy/numpy/blob/529c61bdf69356b4a428d48bb5bc72c2417ed0cf/numpy/core/fromnumeric.py#L1965-L1969

```python
@array_function_dispatch(_shape_dispatcher)
def shape(a):
    """
    ...
    """
    try:
        result = a.shape
    except AttributeError:
        result = asarray(a).shape
    return result
```

This would improve compatibility in support of the ongoing `__array_funtction__` work.

**Comment by leofang** [comment]
Related: To make NEP-18 work we also need `cupy.ndim` (#3060).

**Comment by kmaehashi** [comment]
`cupy.shape` is implemented in v8.0.0b4. #3229

**Comment by beckernick** [comment]
My mistake. I completely missed this. Thanks @kmaehashi !
